tests/unit/test_tf_estimator.py

```diff
@@ -305,6 +308,49 @@ def test_create_model_with_optional_params(sagemaker_session):
     assert model.vpc_config == vpc_config
 
 
+@patch("sagemaker.tensorflow.estimator.TensorFlow._create_tfs_model")
+def test_transformer_creation_with_endpoint_type(create_tfs_model, sagemaker_session):
+    tf = TensorFlow(
+        entry_point=SCRIPT_PATH,
+        role=ROLE,
+        sagemaker_session=sagemaker_session,
+        train_instance_count=INSTANCE_COUNT,
+        train_instance_type=INSTANCE_TYPE,
+    )
+
+    tf.latest_training_job = _TrainingJob(sagemaker_session, JOB_NAME)
+    transformer = tf.transformer(INSTANCE_COUNT, INSTANCE_TYPE, endpoint_type="tensorflow-serving")
+    assert isinstance(transformer, Transformer)
+    assert transformer.sagemaker_session == sagemaker_session
+    assert transformer.instance_count == INSTANCE_COUNT
+    assert transformer.instance_type == INSTANCE_TYPE
+    assert tf.script_mode is True
+    assert tf._script_mode_enabled() is True
+    create_tfs_model.assert_called_once()
```

Comment:
You can do:

```python
create_model.assert_called_with(endpoint_type="tensorflow-serving")
```

Reply:
In this approach, we added transformer() function with endpoint_type parameter at TF estimator class. If endpoint_type=="tensorflow-serving", we change var **script_mode** to True which will return TF script mode + TF Serving. Otherwise, it will use TF legacy/framework mode.

Thus, we don't actually pass **endpoint_type** parameter to transformer() in the Framework base class: https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/estimator.py#L1030 in which **create_model(role=role, model_server_workers=model_server_workers)** is called with only two fixed parameters. So the assertion create_model.assert_called_with(endpoint_type="tensorflow-serving") is failed. 

If we really need to avoid patching a private method, i can override transformer() method from Framework base class to make endpoint_type parameter available.